src/ui-target.c

```diff
@@ -885,22 +883,43 @@ static void load_path(u16b path_n, struct loc *path_g, wchar_t *c, int *a)
 	Term_fresh();
 }
 
+/**
+ * Return true if the object pile contains the player's tracked object
+ */
+static bool pile_is_tracked(const struct object *obj) {
+	for (struct object *o = obj; o != NULL; o = o->next) {
+		if (player->upkeep->object == o) {
+			return true;
```

Comment:
I'm not 100% confident I've gotten the right predicates, but they seem to work.
* `pile_is_tracked` makes sure the player's tracked item is in the given pile so they can't ignore their last-tracked item if the cursor isn't still over the pile. It does what it says, and I think that's sufficient for something you can ignore.
* `pile_has_known` checks for the presence of an item the player could ignore, to inform the help menu. It looks for a known item (!is_unknown). Not sure if there are edge cases.

Reply:
This looks right to me.